### PR TITLE
Create bpmn class

### PR DESCRIPTION
**Pull request by tklaback**
This branch adds the following changes:

- Bpmn class that has a list of processes.
- Process class that holds a set of bpmn elements (vertices) such as gateways, tasks, etc. in the bpmn graph. A process also has a set of flows (edges) and the adjacency list of the graph.
- All Bpmn elements (e.g. Acitivities, Flows, Gateways) inherit from the base Bpmn element class.
- All non-edge elements (activities, gateways, events) inherit from the Node class (which, of course, inherits from Bpmn element).
- All edge (flow) elements inherit from the Flow base class, which inherits from Bpmn element.
- Function called get_bpmn_from_xml(xml: str) that returns a Bpmn object.

### CODE REVIEW

File: src/bpmncwpverify/bpmn/BPMN.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        super().__init__(element)
+
+
+# Process class
+class Process(BpmnElement):
+    def __init__(self, element: Element):
+        super().__init__(element)
+        self.elements: Dict[str, Node] = {}
+        self.flows: Dict[str, Flow] = {}
+        self.adj_list: Dict[str, List[str]] = {}
```
Comment (ericmercer) [inline src/bpmncwpverify/bpmn/BPMN.py:103]: If `adj_list` is redundant, then let's remove it in a new revision.

File: src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                process.elements[source_ref].out_flows.append(flow)
+                # update target node's in flows array
+                process.elements[target_ref].in_flows.append(flow)
+
+
+def _traverse_process(process_element: Element) -> Process:
+    process = Process(process_element)
+
+    for element in process_element:
+        tag_local = element.tag.partition("}")[2]
```
Comment (ericmercer) [inline src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py:63]: Can `element.tag.partition("}")[2]` to a constant? Or a function? Give it a name too that reflects what it does.

File: src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    Process,
+)
+
+
+################
+# Constants
+################
+from bpmncwpverify.constants import NAMESPACES
+
+TAG_CLASS_MAPPING = {
```
Comment (tklaback) [inline src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py:19]: rename this

File: src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        for element_id, outgoing_ids in process.adj_list.items():
+            element_instance = process.elements[element_id]
+            name = element_instance.name
+            print(f"  Element ID: {element_id}, Name: {name}")
+            print("    Outgoing to:")
+            for target_id in outgoing_ids:
+                target_element = process.elements.get(target_id)
+                target_name = target_element.name if target_element else "Unknown"
+                print(f"      Element ID: {target_id}, Name: {target_name}")
+        print()
```
Comment (ericmercer) [inline src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py:94]: Should this be the `Bpmn.__str__` method?

File: src/bpmncwpverify/bpmn/BPMN.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class Bpmn:
```
Comment (ericmercer) [inline src/bpmncwpverify/bpmn/BPMN.py:106]: Add a static method `Bpmn.from(xml: str)`?

**Review by ericmercer — CHANGES_REQUESTED** [review]
Left some comments that talk about what we discussed.

**Pull request merged**